webapi/src/main/scala/org/knora/webapi/responders/v2/OntologyResponderV2.scala

```diff
@@ -1108,30 +1275,563 @@ class OntologyResponderV2 extends Responder {
+                        OntologyConstants.KnoraBase.FileValueClasses.contains(objectClassConstraint.toString) ||
+                        objectClassConstraint.toString == OntologyConstants.KnoraBase.LinkValue) {
+                        throw BadRequestException(s"Invalid object class constraint for value property: ${objectClassConstraint.toOntologySchema(ApiV2WithValueObjects)}")
+                    }
+                }
+
+                allSuperPropertyIris: Set[SmartIri] = internalPropertyDef.subPropertyOf.flatMap {
+                    superPropertyIri => cacheData.subPropertyOfRelations.getOrElse(superPropertyIri, Set.empty[SmartIri])
+                }
+
+                // Check that the subject class, if provided, is a subclass of the subject classes of the base properties.
+
+                _ <- maybeSubjectClassConstraint match {
+                    case Some(subjectClassConstraint) =>
+                        checkPropertyConstraint(
+                            newInternalPropertyIri = internalPropertyIri,
+                            constraintPredicateIri = OntologyConstants.KnoraBase.SubjectClassConstraint.toSmartIri,
+                            constraintValueInNewProperty = subjectClassConstraint,
+                            allSuperPropertyIris = allSuperPropertyIris
+                        )
+
+                    case None => FastFuture.successful(())
+                }
+
+                // Check that the object class is a subclass of the object classes of the base properties.
+
+                _ <- checkPropertyConstraint(
+                    newInternalPropertyIri = internalPropertyIri,
+                    constraintPredicateIri = OntologyConstants.KnoraBase.ObjectClassConstraint.toSmartIri,
+                    constraintValueInNewProperty = objectClassConstraint,
+                    allSuperPropertyIris = allSuperPropertyIris
+                )
+
+                // Add the property (and the link value property if needed).
+
+                currentTime: Instant = Instant.now
+
+                updateSparql = queries.sparql.v2.txt.createProperty(
+                    triplestore = settings.triplestoreType,
+                    ontologyNamedGraphIri = internalOntologyIri,
+                    ontologyIri = internalOntologyIri,
+                    propertyDef = internalPropertyDef,
+                    maybeLinkValuePropertyDef = maybeLinkValuePropDef,
+                    lastModificationDate = createPropertyRequest.lastModificationDate,
+                    currentTime = currentTime
+                ).toString()
+
+                _ <- (storeManager ? SparqlUpdateRequest(updateSparql)).mapTo[SparqlUpdateResponse]
+
+                // Check that the update was successful.
+
+                _ <- checkOntologyLastModificationDateAfterUpdate(internalOntologyIri = internalOntologyIri, expectedLastModificationDate = currentTime)
```

Comment:
Is this check for the last modification date sufficient? Or would you have to check in a more extensive manner?

I guess the whole transaction (`src/main/twirl/queries/sparql/v2/createProperty.scala.txt`) would have been rejected if there had been any error.

Reply:
Ideally, the property would be read back from the triplestore to check that it was stored correctly. But this would require refactoring the SPARQL that queries ontology data, as well as the code in `OntologyResponderV2` that reads the query results. It was never designed to read just one property or one class. I would like to do this later, changing the SPARQL queries to use CONSTRUCT instead of SELECT, and refactoring the Scala code to handle just one property or all the properties in the triplestore, similarly to what you did for reading resources in v2.

Reply:
ok, that's fine for me

we can put that in an issue and do it later

https://de.wiktionary.org/wiki/morgen,_morgen,_nur_nicht_heute,_sagen_alle_faulen_Leute :-)